storybookjs/storybook
# Angular: Refactor angular server

Issue: not yet. but if all goes well i would like to eventually allow storybook to be more "Angular friendly". And to use an angular builder like in NX to simplify the use of storybook in angular

## What I did

Rework the code of `angular-cli_config` and `angular-cli_utils` to add future features 
Improvement:
- Add test for different workspace configuration
These new tests allow to make sure of the compatibility with angular-cli 🤞. And to avoid (like last time) to update the supported angular version without failing 

- Use angular core to read the workspace instead of doing it by hand
- Use angular-cli to read the tsconfig
- Redesigned the code to get out the main steps + added error handling
- Express more clearly the webpack config from angular-cli
- Clarification of the logs
- Improvement of the types with those of angular-cli

I find that it was not easy to understand the code before. I hope to improve it. but if there are still unclear points, do not hesitate to ask questions or suggest improvements. 

## Test campagne

To avoid regressions I added tests before. And check that after refactoring the tests still work without major changes

Step : 
+ Create a new project (`npx @angular/cli@6 new angular-v6`) 
+ Add storybook (`npx sb init`)
+ cp app/angular `dist` in node_modules of the new project
+ Start storybook

- [x] Angular 6
- [x] Angular 9
- [x] Angular 10
- [x] Angular 11
- [x] angular with NX (only with angular) (I didn't succeed in making a new project with a wokspace.json 🤷‍♂️ )

(Skip 7 and 8 🙈 )

⚠️ The goal is to not make any regression !. If you have any ideas for tests, don't hesitate to do them or ask me  🙏 

## How to test

- Is this testable with Jest or Chromatic screenshots? `Yes`
- Does this need a new example in the kitchen sink apps? `Yes`
- Does this need an update to the documentation? `Yes`
